src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,143 @@
+package duke.util;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The class that stores and modifies Tasks of the software.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * Constructor of TaskList.
+     * It instantiates the TaskList object with initial Tasks passed.
+     * @param tasks Initial tasks to be stored in TaskList.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Constructor of TaskList.
+     * It instantiates an empty TaskList object.
+     */
+    public TaskList() {
+        this(new ArrayList<>());
+    }
+
+    /**
+     * Add task to the task list.
+     * @param task The task to add.
+     * @return String of messages to be printed by Ui object after adding.
+     */
+    public String addTask(Task task) {
+        StringBuilder msg = new StringBuilder();
+        this.tasks.add(task);
+        msg.append("Got it. I've added this task:\n");
+        msg.append("\t" + task + "\n");
+        msg.append("Now you have " + tasks.size() + " tasks in the list.\n");
+
+        return msg.toString();
+    }
+
+    /**
+     * Delete task at the specified index.
+     * @param idx The index of the task to be deleted.
+     * @return String of messages to be printed by Ui object after deleting.
+     * @throws DukeException When the index is invalid.
+     */
+    public String deleteTask(int idx) throws DukeException {
+        try {
+            StringBuilder msg = new StringBuilder();
+            Task curr = this.tasks.remove(idx-1);
```

Comment:
Under advanced coding standard violations, it is said not to use 'this' for fields if it is not necessary. Can consider removing this here and other parts of code perhaps.

Reply:
Hey @bernardwan, thanks for pointing that out, I will remove redandunt `this`.